java/ql/test/query-tests/security/CWE-502/A.java

```diff
@@ -21,20 +21,37 @@ public Object deserialize2(Socket sock) throws java.io.IOException, ClassNotFoun
     return in.readUnshared(); // $unsafeDeserialization
   }
 
-  public Object deserialize3(Socket sock) throws java.io.IOException {
+  public Object deserialize3(Socket sock) throws java.io.IOException, ClassNotFoundException {
```

Comment:
Should we test unsafe `resolveClass` implementations as well? Something like:

```java
protected Class<?> resolveClass(ObjectStreamClass osc) throws java.io.IOException, ClassNotFoundException {
  return super.resolveClass(osc);
}
```

Reply:
Nice catch! Thanks! https://github.com/github/codeql/pull/13358/commits/5ea5ca42a4d124897c0d1f2a0398ad74534d69e7